forms-flow-api/src/formsflow_api/services/form_process_mapper.py

```diff
@@ -282,18 +282,47 @@ def check_tenant_authorization_by_formid(form_id: int, **kwargs) -> int:
             raise BusinessException(BusinessErrorCode.PERMISSION_DENIED)
         return
 
-    def _get_form(self, form_name: str, scope_type: str, form_id: str = None) -> dict:
+    def _get_form(  # pylint: disable=too-many-arguments
+        self,
+        title_or_path: str,
+        scope_type: str,
+        form_id: str = None,
+        description: str = None,
+        tenant_key: str = None,
+    ) -> dict:
         """Get form details."""
         try:
-            current_app.logger.info(f"Fetching form : {form_name}")
+            current_app.logger.info(f"Fetching form : {title_or_path}")
             formio_service = FormioService()
             form_io_token = formio_service.get_formio_access_token()
-            if form_id is None:
-                form_json = formio_service.get_form_by_path(form_name, form_io_token)
-            else:
+            if form_id:
                 form_json = formio_service.get_form_by_id(form_id, form_io_token)
+            else:
+                form_json = formio_service.get_form_by_path(
+                    title_or_path, form_io_token
+                )
+            if form_json:
+                form_json.pop("_id", None)
+                form_json.pop("machineName", None)
+                # In a (sub form)connected form, the workflow provides the form path,
+                # and the title is obtained from the form JSON
+                title_or_path = (
+                    form_json.get("title", "") if scope_type == "sub" else title_or_path
+                )
+                # Remove 'tenantkey-' from 'path' and 'name'
+                if current_app.config.get("MULTI_TENANCY_ENABLED"):
+                    tenant_prefix = f"{tenant_key}-"
+                    form_path = form_json.get("path", "")
+                    form_name = form_json.get("name", "")
+                    current_app.logger.info(f"Removing tenant key from path: {form_path} & name: {form_name}")
+                    if form_path.startswith(tenant_prefix):
+                        form_json["path"] = form_path[len(tenant_prefix) :]
+
+                    if form_name.startswith(tenant_prefix):
+                        form_json["name"] = form_name[len(tenant_prefix) :]
             return {
-                "formTitle": form_name,
+                "formTitle": title_or_path,
+                "formDescription": description,
```

Comment:
what is scope type?

Reply:
why we keeping title or path can we keep else title or path right ?

Reply:
scope refers main or sub form/workflow. 
title_or_path argument accept can have title in case of main form & path in case of subform/connected form(We'll get pathname from workflow). 
![image](https://github.com/user-attachments/assets/1fb309c7-8174-4295-82e1-62a43f25e72e)
![image](https://github.com/user-attachments/assets/135c85a5-c07a-4517-b096-406086d6d7c7)